tests/reqRep.cpp

```diff
@@ -0,0 +1,238 @@
+                      BOOST_CHECK_EQUAL(got, expected);
+
+                      handled = true;
+                      return test::Echo("Jumped over the lazy dog").toBinary();
+                  });
+
+    BOOST_CHECK(!handled);
+    server.receive(TIMEOUT);
+    BOOST_CHECK(handled);
+}
+
+bool running;
+};
+}
+
+BOOST_AUTO_TEST_CASE(serializable)
+{
+    test::Echo echo("The quick brown fox");
+
+    zeroeq::Server server(zeroeq::NULL_SESSION);
+    zeroeq::Client client(zeroeq::URI(server.getURI()));
+
+    std::thread thread([&] { runOnce(server, echo); });
+
+    bool handled = false;
+    client.get(echo.getTypeIdentifier(), [&](const void* data,
+                                             const size_t size) {
+        BOOST_CHECK(data);
+        BOOST_CHECK(!handled);
+
+        const auto got = test::Echo::create(data, size);
+        BOOST_CHECK_EQUAL(got.getMessage(), "Jumped over the lazy dog");
+        handled = true;
+    });
+
+    BOOST_CHECK(!handled);
+    client.receive(TIMEOUT);
+    BOOST_CHECK(handled);
+}
+
+#if 0
+BOOST_AUTO_TEST_CASE(no_receive)
+{
+    zeroeq::Server server(zeroeq::URI("1.2.3.4:1234"));
+    BOOST_CHECK(!server.receive(100));
+}
+
+BOOST_AUTO_TEST_CASE(subscribe_to_same_session_zeroconf)
+{
+    if (!servus::Servus::isAvailable() || getenv("TRAVIS"))
+        return;
+
+    zeroeq::Client client(test::buildUniqueSession());
+    BOOST_CHECK_NO_THROW(zeroeq::Server server(client.getSession()));
+}
+
+BOOST_AUTO_TEST_CASE(subscribe_to_different_session_zeroconf)
+{
+    if (!servus::Servus::isAvailable() || getenv("TRAVIS"))
+        return;
+
+    zeroeq::Client client(test::buildUniqueSession());
+    BOOST_CHECK_NO_THROW(zeroeq::Server server(client.getSession() + "bar"));
+}
+
+BOOST_AUTO_TEST_CASE(no_receive_zeroconf)
+{
+    if (!servus::Servus::isAvailable() || getenv("TRAVIS"))
+        return;
+
+    zeroeq::Server server(test::buildUniqueSession());
+    BOOST_CHECK(!server.receive(100));
+}
+
+BOOST_AUTO_TEST_CASE(zeroconf)
+{
+    if (!servus::Servus::isAvailable() || getenv("TRAVIS"))
+        return;
+
+    zeroeq::Client client(test::buildUniqueSession());
+    zeroeq::Server noServer(client.getSession());
+    zeroeq::detail::Sender::getUUID() =
+        servus::make_UUID(); // different machine
+    zeroeq::Server server(client.getSession());
+
+    BOOST_CHECK(server.subscribe(test::Echo::IDENTIFIER(),
+                                 zeroeq::EventPayloadFunc(&test::onEchoEvent)));
+    BOOST_CHECK(
+        noServer.subscribe(test::Echo::IDENTIFIER(),
+                           zeroeq::EventPayloadFunc(&test::onEchoEvent)));
+
+    bool received = false;
+    for (size_t i = 0; i < 20; ++i)
+    {
+        BOOST_CHECK(client.publish(test::Echo(test::echoMessage)));
+
+        BOOST_CHECK(!noServer.receive(100));
+        if (server.receive(100))
+        {
+            received = true;
+            break;
+        }
+    }
```

Comment:
I'm confused. Isn't this for request/reply tests? Why Client has a method called publish? Why servers has a subscribe method? This doesn't resemble the feature spec at all.

I think I see. You just copied the test from pubSub and "commented" this part. If the partial commits don't make sense it's a waste of time to look into them, but looking at the full pull request at once is really hard because of its size. What should I expect from the rest of the commits and what can I expect to be less time consuming? I was already feeling this PR was going to be hard to review and this commit hasn't improved that feeling.